gradio/blocks.py

```diff
@@ -1299,26 +1292,40 @@ def reverse(text):
                     )
         utils.launch_counter()
 
+        self.is_colab = utils.colab_check()
+        self.share = (
+            share
+            if share is not None
+            else True
+            if self.is_colab and self.has_any_queue
```

Comment:
Why do you need this additional `self.has_any_queue` parameter? Isn't `self.queue_enabled` enough?

Reply:
You're right, fixed